### ISSUE TITLE
Change Wallet Encryption `CypherMode`

### ISSUE THREAD

**Issue by cschuchardt88** [neo-project/neo#3140]
**Summary or problem description**
AES with ECB should not be used as a mode for encryption. It has dangerous weaknesses. Data is encrypted the same way every time meaning the same plaintext input will always produce the same ciphertext. This makes encrypted messages vulnerable to replay attacks.

**Do you have any solution you want to propose?**
Use a different `CypherMode`.

**Where in the software does this update applies to?**
- Wallets

**Comment by Jim8y** [comment]
I will check if we have a wallet version in there.

**Comment by roman-khimov** [comment]
This comes from NEP-2 in fact (https://github.com/neo-project/proposals/blob/master/nep-2.mediawiki) and yeah, some years ago I had a similar question. The thing is -- we encrypt a key here and that's just two AES blocks. Two different (with very high probability) AES blocks. Which to me means ECB here is not a problem. At least not something worth changing NEP-2 standard that everyone knows how to work with since Legacy days.

**Comment by cschuchardt88** [comment]
> This comes from NEP-2 in fact (https://github.com/neo-project/proposals/blob/master/nep-2.mediawiki) and yeah, some years ago I had a similar question. The thing is -- we encrypt a key here and that's just two AES blocks. Two different (with very high probability) AES blocks. Which to me means ECB here is not a problem. At least not something worth changing NEP-2 standard that everyone knows how to work with since Legacy days.

Yes, it's not a big deal. Seeing how we don't use it over the network. Are we limited to a certain length of bytes? If not we should change the `CypherMode`. As it sits now; you could crack it the same way as you would with a hash.

**Comment by vang1ong7ang** [comment]
I suggest to improve and keep the compatibility since Tron was criticized by similar issues years ago

**Comment by cschuchardt88** [comment]
This needs to be updated.

**Comment by shargon** [comment]
We can have a version

**Comment by vncoelho** [comment]
> Do we need to update `NEP-6`?

"Rationale
The wallet standard should consider both security and cross-platform compatibility. For security, we require the implementations to use the [NEP-2](https://github.com/neo-project/proposals/blob/master/nep-2.mediawiki) mechanism to encrypt or decrypt the private keys. For cross-platform, we use the JSON format to describe the wallet files, so that the contents of the wallet files can be easily recognized on each platform."

In NEP-6 we refer to NEP-2.

We need a new NEP similar to NEP-2 and a minor update in this section of NEP-6 that will maybe require a new NEP as well.

**Comment by roman-khimov** [comment]
I still think that it's not a problem worth fixing. It'll require quite some effort (and it's not just about core, it's about all of the code that deals with NEP-6 and NEP-2) and the benefit is close to zero.

**Comment by vncoelho** [comment]
> I still think that it's not a problem worth fixing. It'll require quite some effort (and it's not just about core, it's about all of the code that deals with NEP-6 and NEP-2) and the benefit is close to zero.

I think the same for now.
I would set this for a future version.
